transformations/disability_transformation/transformation.py

```diff
@@ -0,0 +1,213 @@
+                new_str += st[j:i-1]
+                j = i
+            new_str += st[j:]
+            return new_str
+        
+        elif indices[-1] < l-1:
+            j = 0
+            for i in indices:
+                new_str += st[j:i-1]
+                j = i
+            notlast = indices[-1]
+            new_str += st[notlast:]
+            return new_str
+          
+    elif len(indices) == 0:
+        return st
+    else:
+        return st
+
+#CAPITALIZATION
+def preserve_capitalization(original, transformed):
+    if original[0].isupper():
+        transformed = transformed.capitalize()
+    else:
+        return original
+    return transformed
+
+def get_index(wl,n):
+    indices = [i for i, x in enumerate(wl) if x == n]
+    return indices
+
+#ASSUMING FOR NAME IN TEXT
+
+def placement(index_of_dis, wl,pl,input,disability_names,name):
+
+    text = input.lower()
+    wl,pl = postag(text)
+    index_of_dis = get_index(wl,name)
+    max_len = len(wl)
+    #FOR EACH VALUE OF INDEX
+    for i in index_of_dis: 
+        print("For Occurence", i)
+        print("For Words Less than Maximum Length:")
+      
+        if i < (max_len-1):
+          print("Words in between")
+          if pl[i+1] == 'NOUN': 
+              s = ' '.join(wl)
+              text = s
+   
+          elif pl[i+1]!='NOUN':
+              s = ' '.join(wl) #convert to a string and replace with disability name
+              namew = wl[i]
+              wl[i] = disability_names[namew]
+              text = ' '.join(wl)
+
+    #last word    
+        elif i >= (max_len-1):
+            print("For Words At Maximum Length:") 
+            namew = wl[i]
+            wl[i] = disability_names[namew]
+            text = ' '.join(wl)
+   
+        else:
+            s = ' '.join(wl)
+            text = s
+
+        text = preserve_capitalization(input, text)
+    return text    
+                          
+
+def different_ability(input, disability_names):
+  #lowercase
+    text = input.lower()
+
+    for name in disability_names.keys():
+        if name in text:
+            #GET POS TAGS
+            wl,pl = postag(text)
+            #MAX LENGTH OF LIST
+            max_len = len(wl)
+            #GET ALL OCCURENCES OF NAME
+            indices = get_index(wl,name)
+            #PLACEMENT
+            textp = placement(indices,wl,pl,input,disability_names,name) 
+            #restore end formatting
+            text = cleanup(textp)
+    return text
+
+
+class DifferentAbilityTransformation(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION
+    ]
+    languages = ["en"]
+
+    def __init__(self, seed=0, max_outputs=1):
+        super().__init__(seed, max_outputs=max_outputs)
+        
+       
+        self.disability_names = disability_names
+
+    def generate(self, sentence: str):
+
+      return [different_ability(sentence, self.disability_names)]
+    
+  
+  
+  
+if __name__ == '__main__':
```

Comment:
This can be commented or removed.

Reply:
I have made the following changes: removed excessive new lines, added spaces after commas and removed this end part.
 Thanks.